igvc_rviz_plugins/src/WaypointPanel.cpp

```diff
@@ -0,0 +1,61 @@
+#define _USE_MATH_DEFINES
+
+#include <igvc_rviz_plugins/WaypointPanel.h>
+#include <QVBoxLayout>
+#include <pluginlib/class_list_macros.h>
+#include <tf/LinearMath/Matrix3x3.h>
+
+namespace igvc_rviz_plugins {
+
+WaypointPanel::WaypointPanel(QWidget *parent)
+  : rviz::Panel(parent)
+{
+    ros::NodeHandle handle;
+
+    QLabel *label = new QLabel("0");
+
+    waypoint_subscriber = handle.subscribe<geometry_msgs::PointStamped>("/waypoint", 1, boost::bind(&WaypointPanel::waypoint_callback, this, _1, label));
+    robot_position_subscriber = handle.subscribe<nav_msgs::Odometry>("/odometry/filtered", 1, boost::bind(&WaypointPanel::robot_position_callback, this, _1, label));
+
+    QVBoxLayout *layout = new QVBoxLayout;
+    layout->addWidget(label);
+    setLayout(layout);
+}
+
+void WaypointPanel::waypoint_callback(const geometry_msgs::PointStampedConstPtr &msg, QLabel *label) {
+    way_x = msg->point.x - robot_x;
+    way_y = msg->point.y - robot_y;
+}
+
+void WaypointPanel::robot_position_callback(const nav_msgs::OdometryConstPtr &msg, QLabel *label) {
+    double distance = std::hypot(way_x, way_y);
+    double angle = atan2(way_y, way_x) * 180 / M_PI;
```

Comment:
If you can, give this a more descriptive name. Here, that could be as simple as `globalAngle` or `angleBetweenPoints`. Pick a name you like that would explain to a new reader of this code that this is not just any angle but an angle measurement that means something specific.